server/controllers/entities/lib/validate_and_format_claim.ts

```diff
@@ -1,8 +1,49 @@
+import { externalIdsProperties } from '#controllers/entities/lib/properties/properties_values_constraints'
+import { validateSnakValueSync } from '#controllers/entities/lib/validate_claim_sync'
+import { isNonEmptyPlainObject } from '#lib/boolean_validations'
+import { newError } from '#lib/error/error'
+import { arrayIncludes, objectEntries } from '#lib/utils/base'
+import { isClaimObject } from '#models/entity'
+import type { InvClaimObject } from '#server/types/entity'
 import { validateProperty } from './properties/validations.js'
-import { validateAndFormatClaimValue } from './validate_and_format_claim_value.js'
+import { validateAndFormatClaimValue, type ValidateAndFormatClaimValueParams } from './validate_and_format_claim_value.js'
 
-export async function validateAndFormatClaim (params) {
-  const { property } = params
+export async function validateAndFormatClaim (params: ValidateAndFormatClaimValueParams) {
+  const { property, newClaim } = params
   validateProperty(property)
+  if (newClaim != null && isClaimObject(newClaim)) {
+    validateAndFormatReferences(newClaim)
+  }
   return validateAndFormatClaimValue(params)
 }
+
+function validateAndFormatReferences (claim: InvClaimObject) {
+  const { references } = claim
+  if (!references) return
+  if (!(references instanceof Array)) {
+    throw newError('invalid reference array', 400, { claim, references })
```

Comment:
Error message could be more explicit imo:

```suggestion
    throw newError('invalid references value, should be an array', 400, { claim, references })
```

Reply:
done in b509db6f8